src/types/varnum.rs

```diff
@@ -0,0 +1,195 @@
+//! Protocol Buffer Varints.
+
+use std::old_io::IoResult;
+use std::iter::range_step;
+
+use packet::Protocol;
+
+/// Protocol Buffer Varint, encoding a two's complement signed 32-bit integer.
+#[derive(Copy, Debug)]
+pub struct Varint;
```

Comment:
Protobuf uses `varint`, capitalized only at the beginning of a sentence. In my opinion, the `VarInt` spelling makes more sense because of the origin of the name. The unofficial [protocol documentation](http://wiki.vg/Protocol) also uses this form.